core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -222,6 +227,24 @@ void setSpecId(int specId) {
     this.partitionSpecId = specId;
   }
 
+  @Override
+  public Long dataSequenceNumber() {
+    return this.dataSequenceNumber;
```

Comment:
minor: We rarely use explicit `this` whenever accessing a var, just when setting.